public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -58,6 +58,8 @@ export class PrometheusDatasource extends DataSourceApi<PromQuery, PromOptions>
   withCredentials: any;
   metricsNameCache = new LRU<string, string[]>(10);
   interval: string;
+  safeInterval: number;
+  currentInterval: number;
```

Comment:
i don't think it is a good idea to store these as instance-variables.

both change with every query (they both depend on the currently selected time-range i think), so i recommend to just calculate them when it is needed. otherwise the code becomes vulnerable to problems when two queries are running at the same time.

for example, when accessing from PromQueryField like:
```
    const queryHints = datasource.getQueryHints(query, result, datasource.safeInterval);
```
a better approach would be something like:
```
    const queryHints = datasource.getQueryHints(query, result, datasource.getSafeInterval(currentTimeRange));
```

Reply:
That's a great solution! Thank you @gabor